plugins/polio/api/vaccines/supply_chain.py

```diff
@@ -138,27 +167,55 @@ class Meta(NestedVaccinePreAlertSerializerForPost.Meta):
 
     def validate(self, attrs: Any) -> Any:
         # at least one of the other fields must be present
-        if not any(key in attrs.keys() for key in NestedVaccinePreAlertSerializerForPost.Meta.fields):
-            raise serializers.ValidationError("At least one of the fields must be present.")
+        if not any(
+            key in attrs.keys()
+            for key in NestedVaccinePreAlertSerializerForPost.Meta.fields
+        ):
+            raise serializers.ValidationError(
+                "At least one of the fields must be present."
+            )
 
         validated_data = super().validate(attrs)
-        if "PO" in validated_data.get("po_number", "") or "po" in validated_data.get("po_number", ""):
+        if "PO" in validated_data.get("po_number", "") or "po" in validated_data.get(
+            "po_number", ""
+        ):
             raise serializers.ValidationError("PO number should not be prefixed")
 
         # Get current object
         current_obj = VaccinePreAlert.objects.get(id=attrs["id"])
 
         # Check if any values are actually different
-        has_changes = False
+        is_different = False
         for key in attrs.keys():
-            if hasattr(current_obj, key) and getattr(current_obj, key) != attrs[key]:
-                has_changes = True
+            if key == "document":
+                # Skip if no new document is being uploaded
+                if not attrs[key]:
+                    continue
+
+                # If there's no existing document but we're uploading one
+                if not getattr(current_obj, key):
+                    is_different = True
+                    setattr(current_obj, key, attrs[key])
```

Comment:
```suggestion
                    current_obj.document = attrs[key]
```

I'm not going to point it out everywhere, but it's valid for the whole PR.